packages/core/contracts/optimistic-assertor/implementation/examples/PredictionMarket.sol

```diff
@@ -0,0 +1,261 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "../../../common/implementation/AddressWhitelist.sol";
+import "../../../common/implementation/AncillaryData.sol";
+import "../../../common/implementation/ExpandedERC20.sol";
+import "../../../oracle/implementation/Constants.sol";
+import "../../../oracle/interfaces/FinderInterface.sol";
+import "../../interfaces/OptimisticAssertorInterface.sol";
+import "../../interfaces/OptimisticAssertorCallbackRecipientInterface.sol";
+
+// This contract allows to initialize prediction markets each having a pair of binary outcome tokens. Anyone can mint
+// and burn the same amount of paired outcome tokens for the default payout currency. Trading of outcome tokens is
+// outside the scope of this contract. Anyone can assert 3 possible outcomes (outcome 1, outcome 2 or split) that is
+// verified through Optimistic Asserter. If the assertion is resolved true then holders of outcome tokens can settle
+// them for the payout currency based on resolved market outcome.
+contract PredictionMarket is OptimisticAssertorCallbackRecipientInterface {
+    using SafeERC20 for IERC20;
+
+    struct Market {
+        bool resolved; // True if the market has been resolved and payouts can be settled.
+        bytes32 assertedOutcomeId; // Hash of asserted outcome (outcome1, outcome2 or splitOutcome).
+        ExpandedIERC20 outcome1Token; // ERC20 token representing the value of the first outcome.
+        ExpandedIERC20 outcome2Token; // ERC20 token representing the value of the second outcome.
+        uint256 reward; // Reward available for asserting true market outcome.
+        uint256 requiredBond; // Expected bond to assert market outcome (OA can require higher bond).
+        bytes outcome1; // Short name of the first outcome.
+        bytes outcome2; // Short name of the second outcome.
+        bytes description; // Description of the market.
+    }
+
+    struct AssertedMarket {
+        address asserter; // Address of the asserter used for reward payout.
+        bytes32 marketId; // Identifier for markets mapping.
+    }
+
+    mapping(bytes32 => Market) public markets; // Maps marketId to Market struct.
+
+    mapping(bytes32 => AssertedMarket) public assertedMarkets; // Maps assertionId to AssertedMarket.
+
+    FinderInterface public immutable finder; // UMA protocol Finder used to discover other protocol contracts.
+    IERC20 public immutable currency; // Currency used for all prediction markets.
+    OptimisticAssertorInterface public immutable oa;
+    uint256 public constant assertionLiveness = 7200; // 2 hours.
+    bytes32 public immutable defaultIdentifier; // Identifier used for all prediction markets.
+    bytes public constant splitOutcome = "Unknown"; // Name of the split outcome.
```

Comment:
Maybe we should rename this variable:
`splitOutcome` as I think it works as splitOutcome but also as an unresolvable/unknown etc...
Maybe `unresolvable` or `unknown`?

Reply:
updated to `unresolvable`